rust/tool/src/generation.rs

```diff
@@ -40,10 +40,28 @@ pub struct Generation {
 impl Generation {
     pub fn from_link(link: &GenerationLink) -> Result<Self> {
         let bootspec_path = link.path.join("boot.json");
-        let generation: BootspecGeneration = serde_json::from_slice(
-            &fs::read(bootspec_path).context("Failed to read bootspec file")?,
-        )
-        .context("Failed to parse bootspec json")?;
+        let generation: BootspecGeneration;
+
+        if let Ok(bootspec_data) = fs::read(bootspec_path) {
+            generation = serde_json::from_slice(&bootspec_data)
+                .context("Failed to parse bootspec JSON")
```

Comment:
Would this error message ever be displayed?

Reply:
Can't these two cases that end up in synthesizing be unified by doing `fs::read(bootspec_path).and_then(...).or_else(... synthesize ...)`?

Reply:
Good point, my brain was stuck trying to find a good way

Reply:
I tried something, LMK what you think about it.

Reply:
(we need to call `.context` or something otherwise the Error type will be std::io::Error, we want anyhow::Error)